### PR TITLE
Explain to the compiler that there is an external instantiation of some member variables.

### PR DESCRIPTION
**Pull request by bangerth**
Supersedes #5175.

### CODE REVIEW

File: source/simulator/simulator_signals.cc [hunk 1]
```diff
@@ -96,8 +96,25 @@ namespace aspect
 #define INSTANTIATE(dim) \
   template struct SimulatorSignals<dim>;
 
-
   ASPECT_INSTANTIATE(INSTANTIATE)
 
+  // Now also instantiate static member variables:
+  template
```
Comment (tjhei) [inline source/simulator/simulator_signals.cc:102]: Can you use the ASPECT_INSTANTIATE macro above? This would do the ``dim`` part for you.

**Review by tjhei — COMMENTED** [review]
The testers are not happy with the current solution.

**Comment by bangerth** [comment]
I needed to get far enough in my compile to actually compile this file locally. I think this should now work.

**Comment by bangerth** [comment]
OK, this seems to work now!

**Review by gassmoeller — APPROVED** [review]
Looks good to me, but I will let @tjhei take another look.

**Review by tjhei — APPROVED** [review]

**Pull request merged**